src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+  }
+
+  compare(first, second) {
+    return typeof first;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+    Animal.alive.push(this);
```

Comment:
We can push "_this_" in Animal class.

Reply:
You are repeating `Animal.alive.push(this);` two times. Why don't you write it once in the `Animal` class, and two other classes will inherit it?